hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -397,6 +399,36 @@ public void testLockExceptionUnknownSuccessCommit() throws TException, Interrupt
         .isTrue();
   }
 
+  @Test
+  public void testChangeLockWithAlterTable() throws Exception {
+    Table table = catalog.loadTable(TABLE_IDENTIFIER);
+    HiveTableOperations ops = (HiveTableOperations) ((HasTableOperations) table).operations();
+    TableMetadata base = ops.current();
+    final HiveLock initialLock = ops.lockObject(base);
+
+    AtomicReference<HiveLock> lockRef = new AtomicReference<>();
+    HiveTableOperations spyOps = spy(ops);
+    doAnswer(
+            i -> {
+              lockRef.set(ops.lockObject(i.getArgument(0)));
+              return lockRef.get();
+            })
+        .when(spyOps)
+        .lockObject(any());
```

Comment:
Can we avoid `any` if possible?

Reply:
Yeah, but would you mind elaborate why `any` is bad? The stub here is meant to capture the lock returned by `lockObject`. I don't understand why we care about the argument.

Reply:
If we check the argument, then we also check, that our assumptions about the tests are correct. I recently faced an issue when an unrelated change made the test irrelevant (the methods called with different parameter values), but the tests were green, because `any` were used to match the arguments

Reply:
Thanks for explaining. PR updated accordingly.